repo/fs/fs.go

```diff
@@ -79,6 +83,35 @@ func NewRepo(store cafs.Filestore, fsys qfs.Filesystem, book *logbook.Book, cach
 	return r, nil
 }
 
+// ResolveRef implements the resolve.RefResolver interface
+func (r *Repo) ResolveRef(ctx context.Context, ref *dsref.Ref) error {
+	if r == nil {
+		return resolve.ErrCannotResolveName
+	}
+
+	if ref.Username == "me" {
+		ref.Username = r.profile.Peername
+	}
+
+	match, err := r.GetRef(reporef.RefFromDsref(*ref))
+	if err != nil {
+		return resolve.ErrCannotResolveName
+	}
+
+	// TODO (b5) - repo doens't store IDs yet, breaking the assertion that ResolveRef
+	// will set the ID of a dataset. Need to fix that before we can ship this
```

Comment:
This is strong evidence that putting the same interface onto each subcomponent doesn't make sense. Repo fundamentally doesn't have the same surface area as the top-level resolver. It doesn't know about initID because it doesn't know about logbook.

Reply:
No, this is evidence that `repo.RefStore` is an API we're intentionally letting rot on the vine. It doesn't have the same surface area as the same top level resolver because we're prioritizing work dscache, which _does_ have the the same surface area when it comes to this definition of reference resolution.

This is a deviation from the behavior that `ResolveRef` expects. We can solve this with an error to indicate different behaviour.